bindings/python/triplestore/triplestore.py

```diff
@@ -0,0 +1,608 @@
+by rdflib.Graph, but simplified when possible to make it easy to use.  Some
+important differences:
+- all IRIs are represented by Python strings
+- blank nodes are strings starting with "_:"
+- literals are constructed with Literal()
+
+The module already provides a set of pre-defined namespaces that simplifies
+writing IRIs. For example:
+
+    >>> from triplestore import RDFS, OWL
+    >>> RDFS.subClassOf
+    'http://www.w3.org/2000/01/rdf-schema#subClassOf'
+
+New namespaces can be created using the Namespace class, but are usually
+added with the bind() method:
+
+    >>> from triplestore import Triplestore
+    >>> ts = Triplestore(backend="rdflib")
+    >>> ONTO = ts.bind("onto", "http://example.com/onto#")
+    >>> ONTO.MyConcept
+    'http://example.com/onto#MyConcept'
+
+New triples can added either with the parse() method (for backends that support
+it) or the add() and add_triples() methods.
+
+    # en(msg) is a convinient function for adding english literals.
+    # It is equivalent to ``triplestore.Literal(msg, lang="en")``.
+    >>> from triplestore import en
+    >>> ts.parse("onto.ttl", format="turtle")
+    >>> ts.add_triples([
+    ...     (ONTO.MyConcept, RDFS.subClassOf, OWL.Thing),
+    ...     (ONTO.MyConcept, RDFS.label, en("My briliant ontological concept.")),
+    ... ])
+
+For backends that support it can the triplestore be serialised using
+serialize():
+
+    >>> ts.serialize("onto2.ttl")
+
+A set of convenient functions exists for simple queries, including
+triples(), subjects(), predicates(), objects(), subject_predicates(),
+subject_objects(), predicate_objects() and value().  Except for value(),
+they return the result as generators. For example:
+
+    >>> list(ts.objects(subject=ONTO.MyConcept, predicate=RDFS.subClassOf))
+    ['http://www.w3.org/2002/07/owl#Thing']
+
+The query() and update() methods can be used to query and update the
+triplestore using SPARQL.
+
+Finally Triplestore has two specialised methods add_mapsTo() and
+add_function() that simplify working with mappings.  add_mapsTo() is
+convinient for defining new mappings:
+
+    >>> from triplestore import Namespace
+    >>> META = Namespace("http://onto-ns.com/meta/0.1/MyEntity#")
+    >>> ts.add_mapsTo(ONTO.MyConcept, META.my_property)
+
+It can also be used with DLite and SOFT7 data models.  Here we repeat
+the above with DLite:
+
+    >>> import dlite
+    >>> meta = dlite.get_entity("http://onto-ns.com/meta/0.1/MyEntity")
+    >>> ts.add_mapsTo(ONTO.MyConcept, meta, "my_property")
+
+The add_function() describes a function and adds mappings for its
+arguments and return value(s).  Currently it only supports the Function
+Ontology (FnO).
+
+    >>> def mean(x, y):
+    ...     """Returns the mean value of `x` and `y`."""
+    ...     return (x + y)/2
+
+    >>> ts.add_function(mean, expects=(ONTO.RightArmLength, ONTO.LeftArmLength),
+    ...                 returns=ONTO.AverageArmLength)
+
+
+TODO:
```

Comment:
Probably should turn these TODO's into issues and remove them from the code

Reply:
Agree. Should be done...